src/permutat.cc

```diff
@@ -355,6 +359,14 @@ static Obj ProdPerm(Obj opL, Obj opR)
     degP = degL < degR ? degR : degL;
     prd  = NEW_PERM<Res>( degP );
 
+    // Deal with trivial cases
+    if (degL == 0) {
+        return opR;
+    }
+    if (degR == 0) {
+        return opL;
+    }
```

Comment:
Perhaps move these test *before* creating the now useless temporary object `prd`?